tagstudio/src/qt/widgets/tag_box.py

```diff
@@ -73,13 +93,33 @@ def __init__(
             f"background: #555555;"
             f"}}"
         )
+
         tsp = TagSearchPanel(self.driver.lib)
-        tsp.tag_chosen.connect(lambda x: self.add_tag_callback(x))
+        tsp.tag_chosen.connect(
+            lambda x: (
+                self.add_tag_callback(x),
+                self.tag_entry.clear(),
+            )
+        )
         self.add_modal = PanelModal(tsp, title, "Add Tags")
-        self.add_button.clicked.connect(
-            lambda: (
-                tsp.update_tags(),
-                self.add_modal.show(),
+
+        self.add_button.clicked.connect(self.add_modal.show)
+        self.tag_entry.textChanged.connect(
+            lambda text: (
+                tsp.search_field.setText(text),
+                tsp.update_tags(text),
+                self.tag_completer.setCompletionPrefix(text),
+                self.tag_completer.complete(),
+            )
+        )
+        self.tag_entry.returnPressed.connect(
+            lambda: self.tag_completer.activated.emit(self.tag_entry.text())
```

Comment:
pressing Enter behaves differently than I'd expect in a scenario when no match is found:

Steps to reproduce:

- tags "Archived" and "Favorite" in DB.
- type "Favorited" (with extra `d` on the end), so there no match
- press Enter
- no tag is added, which is expected behaviour, since there was no match (thanks for fixing that)
- the value of the search field is emptied, which is not the expected behaviour when there was no match - right now I might be confused why the tag "Favorite" was not added, since I am convinced I typed there "Favorite". So in the case of no match, I would expect Enter wont empty the value.

does it make sense?